commons/zenoh-codec/src/core/zbuf.rs

```diff
@@ -103,7 +103,7 @@ where
                     reader.read_zslices(len, |s| zbuf.push_zslice(s))?;
                 }
                 super::zslice::kind::SHM_INFO => {
-                    let bytes: Vec<u8> = self.codec.read(&mut *reader)?;
+                    let bytes: alloc::vec::Vec<u8> = self.codec.read(&mut *reader)?;
```

Comment:
I would use `use alloc::vec::Vec` at the beginning of the file.

Reply:
This piece of code is compiled only when `#[cfg(feature = "shared-memory")]`, hence putting `use alloc::vec::Vec` at the beginning would result in a `unused import` warning whenever the `shared-memory` feature is not enabled.
Should I move the import to the top anyways, and add a feature gate to it?

Reply:
Rust seems to complain:
```bash
warning: the item `Vec` is imported redundantly
   --> commons/zenoh-codec/src/core/zbuf.rs:96:13
    |
96  |         use alloc::vec::Vec;
    |             ^^^^^^^^^^^^^^^
    |
   ::: /home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/prelude/mod.rs:119:13
    |
119 |     pub use super::v1::*;
    |             ------------ the item `Vec` is already defined here
    |
    = note: `#[warn(unused_imports)]` on by default
```